homeassistant/components/pilight/__init__.py

```diff
@@ -138,23 +145,23 @@ class CallRateDelayThrottle:
     def __init__(self, hass, delay_seconds: float) -> None:
         """Initialize the delay handler."""
         self._delay = timedelta(seconds=max(0.0, delay_seconds))
-        self._queue: list = []
+        self._queue: list[Callable[[Any], None]] = []
         self._active = False
         self._lock = threading.Lock()
         self._next_ts = dt_util.utcnow()
         self._schedule = functools.partial(track_point_in_utc_time, hass)
 
-    def limited(self, method):
+    def limited(self, method: Callable[_P, Any]) -> Callable[_P, None]:
```

Comment:
We should probably just replace this class by using the Debouncer instead.

Reply:
I'm probably missing something here 🤔
The class is used to throttle a service call. I'm not sure how we can pass the `ServiceCall` argument to the debouncer, since it doesn't allow passing in arguments.

https://github.com/home-assistant/core/blob/6b60fb954197003c05789a6f34e91d2989e091eb/homeassistant/components/pilight/__init__.py#L88-L100

Reply:
Right, forgot about that. :smile: